File: 백준/Silver/10845. 큐/큐.py
front, rear = 0, 0
q = [0] * 10000
for _ in range(int(input())):
    command = input().split()
    if command[0] == 'push':
        q[rear] = int(command[1])
        rear += 1

    elif command[0] == 'pop':
        if front != rear:
            print(q[front])
            front += 1
        else:
            print(-1)

    elif command[0] == 'size':
        print(rear - front)
    elif command[0] == 'empty':
        print(int(rear == front))

    elif command[0] == 'front':
        if rear == front:
            print(-1)
        else:
            print(q[front])
    elif command[0] == 'back':
        if rear == front:
            print(-1)
        else:
            print(q[rear - 1])

File: 백준/Silver/10845. 큐/큐.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX_QUEUE_SIZE 10000
#define MAX_COMMAND_SIZE 10

typedef struct
{
    int arr[MAX_QUEUE_SIZE];
    int front;
    int back;
} Queue;

void push(Queue *q, int x)
{
    q->arr[q->back++] = x;
}

int pop(Queue *q)
{
    return (q->front == q->back) ? -1 : q->arr[q->front++];
}

int size(Queue *q)
{
    return q->back - q->front;
}

int empty(Queue *q)
{
    return (q->front == q->back) ? 1 : 0;
}

int front(Queue *q)
{
    return (q->front == q->back) ? -1 : q->arr[q->front];
}

int back(Queue *q)
{
    return (q->front == q->back) ? -1 : q->arr[q->back - 1];
}

int main()
{
    int N;
    scanf("%d", &N);

    Queue q;
    q.front = 0;
    q.back = 0;

    for (int i = 0; i < N; i++)
    {
        char command[MAX_COMMAND_SIZE];
        scanf("%s", command);

        if (strcmp(command, "push") == 0)
        {
            int x;
            scanf("%d", &x);
            push(&q, x);
        }
        else if (strcmp(command, "pop") == 0)
            printf("%d\n", pop(&q));
        else if (strcmp(command, "size") == 0)
            printf("%d\n", size(&q));
        else if (strcmp(command, "empty") == 0)
            printf("%d\n", empty(&q));
        else if (strcmp(command, "front") == 0)
            printf("%d\n", front(&q));
        else if (strcmp(command, "back") == 0)
            printf("%d\n", back(&q));
    }

    return 0;
}